addons/arsenal/functions/fnc_baseAttachment.sqf

```diff
@@ -0,0 +1,34 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Jonpas, LinkIsGrim
+ * Returns base attachment for CBA scripted attachment
+ *
+ * Arguments:
+ * 0: Attachment <STRING>
+ *
+ * Return Value:
+ * Base attachment <STRING>
+ *
+ * Example:
+ * "CUP_optic_Elcan_SpecterDR_black_PIP" call ace_arsenal_fnc_baseAttachment
+ *
+ * Public: Yes
+ */
+
+params [["_item", "", [""]]];
+
+private _nextCfg = configFile >> "CfgWeapons" >> _item;
+private _switchEntry = ["MRT_SwitchItemPrevClass", "MRT_SwitchItemNextClass"] select (isText (_nextCfg >> "MRT_SwitchItemNextClass"));
+
+while {getNumber (_nextCfg >> "scope") < 2} do {
```

Comment:
Can't this create an infinite loop, if e.g. 2 modes for an attachment, each with `scope = 1`?

Reply:
I can add checks to make sure we're not iterating over the same class twice, but other than that not sure how to handle this

Reply:
Good enough imo.

Reply:
Rewrote `FUNC(baseAttachment)` to use the same method as [CBA_fnc_switchableAttachments](https://github.com/CBATeam/CBA_A3/blob/release/addons/accessory/fnc_switchableAttachments.sqf)

Result from making `ace_acc_pointer_red` loop:

```
12:11:23 [ACE] (arsenal) TRACE: 0 found class: _item=ACE_acc_pointer_green_IR, _x=ACE_acc_pointer_green z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:51
12:11:23 [ACE] (arsenal) TRACE: 0 updating baseAttachment: _xClass=ace_acc_pointer_green_ir, _baseAttachment=ACE_acc_pointer_green z\ace\addons\arsenal\functions\fnc_scanConfig.sqf:321
12:11:23 [ACE] (arsenal) TRACE: 0 looking up base attachment: _item=ace_acc_pointer_red z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:22
12:11:23 [ACE] (arsenal) TRACE: 0 found item with same base class: _item=ACE_acc_pointer_red, count _switchableClasses=1 z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:58
12:11:23 [ACE] (arsenal) TRACE: 0 looking up base attachment: _item=ace_acc_pointer_green z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:22
12:11:23 [ACE] (arsenal) TRACE: 0 found class: _item=ACE_acc_pointer_green, _x=ACE_acc_pointer_green z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:51
12:11:23 [ACE] (arsenal) TRACE: 0 found item with same base class: _item=ACE_acc_pointer_green, count _switchableClasses=2 z\ace\addons\arsenal\functions\fnc_baseAttachment.sqf:58

```